cmd/clusterctl/client/repository/repository_versions.go

```diff
@@ -110,7 +110,7 @@ func latestPatchRelease(ctx context.Context, repo Repository, major, minor *int3
 	}
 
 	if len(versionCandidates) == 0 {
-		return "", errors.New("failed to find releases tagged with a valid semantic version number")
+		return "", errors.New("failed to find releases tagged with a valid semantic version number (metadata.yaml up-to-date?)")
```

Comment:
It's not always out of date, it could be simply that this provider version is not compatible with the chosen capi version.

I'd guess the metadata misses the version thing only happens in one of these three cases? and the message might be misleading in the two cases where we would return it but its unrelated?!

Reply:
If you are new to that, then it is likely that you do not know about metadata.yaml.

What about just adding `(metadata.yaml)`?

But overall it is now no longer very important for me. I just care for other people which are new to that topic.

We can close the PR, if we do not find a simple solution. Maybe this PR (even it is just closed and not merged) helps people to find the right answer :-)